sbin/unit-tests

```diff
@@ -1,189 +1,279 @@
-		export ASAN_OPTIONS="detect_odr_violation=0:alloc_dealloc_mismatch=0:halt_on_error=0:detect_leaks=1:log_path=${ASAN_LOG}:verbosity=1:log_thread=1"
-		export LSAN_OPTIONS="suppressions=$ROOT/tests/memcheck/asan.supp:verbosity=1:log_thread=1"
-		# :use_tls=0
-
-	fi
+#------------------------------------------------------------------------------
+# Run C unit tests
+#------------------------------------------------------------------------------
+run_c_tests() {
+    print_separator
+    echo "# Running C unit tests"
+    C_TESTS_DIR="$(cd $BINROOT/search-community/tests/ctests; pwd)"
+    cd $ROOT_DIR/tests/ctests
+
+    if [[ -z $TEST ]]; then
+        # Run all C tests
+        for test in $(find $C_TESTS_DIR -maxdepth 1 -name "test_*" -type f -print); do
+            run_single_test "$test" ""
+        done
+    elif [[ -f $C_TESTS_DIR/$TEST ]]; then
+        # Run single C test
+        run_single_test "$C_TESTS_DIR/$TEST" ""
+    else
+        echo "Test not found: $TEST in $C_TESTS_DIR"
+    fi
 }
 
-#------------------------------------------------------------------------------------ Arguments
-
-[[ $1 == --help || $1 == help || $HELP == 1 ]] && { help; exit 0; }
-
-OP=
-[[ $NOP == 1 ]] && OP=echo
+#------------------------------------------------------------------------------
+# Run C++ unit tests
+#------------------------------------------------------------------------------
+run_cpp_tests() {
+    print_separator
+    echo "# Running C++ unit tests"
+    CPP_TESTS_DIR="$(cd $BINROOT/search-community/tests/cpptests; pwd)"
+    cd $ROOT_DIR/tests/cpptests
+    TEST_NAME=rstest setup_sanitizer
+
+    # C++ tests are special because they use Google Test framework
+    # which runs multiple tests in a single binary
+    if [[ -z $TEST ]]; then
+        # Run all C++ tests
+        LOG_FILE="${LOGS_DIR}/rstest.log"
+        echo "Running all C++ tests (log: $LOG_FILE)"
+        { $CPP_TESTS_DIR/rstest > "$LOG_FILE" 2>&1; test_result=$?; (( EXIT_CODE |= $test_result )); } || true
+
+        # Parse and display individual test results
+        parse_cpp_test_results "$LOG_FILE"
+    else
+        # Run single C++ test if requested
+        if [[ -f $CPP_TESTS_DIR/rstest ]]; then
+            LOG_FILE="${LOGS_DIR}/rstest_${TEST}.log"
+            echo "Running C++ test: $TEST (log: $LOG_FILE)"
+            { $CPP_TESTS_DIR/rstest --gtest_filter=$TEST > "$LOG_FILE" 2>&1; test_result=$?; (( EXIT_CODE |= $test_result )); } || true
+
+            # Parse and display results
+            parse_cpp_test_results "$LOG_FILE"
+        else
+            echo "C++ test binary not found: $CPP_TESTS_DIR/rstest"
+        fi
+    fi
+}
 
-#--------------------------------------------------------------------------------- Environments
+#------------------------------------------------------------------------------
+# Parse and display C++ test results
+#------------------------------------------------------------------------------
+parse_cpp_test_results() {
```

Comment:
What was the motivation of parsing the already parsed output of the cpp-tests?

Reply:
So now output to console is minimal, only indicates which tests pass and which failed, and full logs are in log file